frontendUiRootImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/VcsRootComboBox.java

```diff
@@ -36,7 +55,13 @@ public VcsRootComboBox(@MinLen(1) List<GitRepository> repositories) {
 
   @Override
   @UIEffect
-  public void updateRepositories(@MinLen(1) List<GitRepository> repositories) {
+  public void updateRepositories() {
+    // GitUtil.getRepositories(project) should never return empty list because it means there's no git repository in an opened
+    // project, so Git Machete plugin shouldn't even be loaded in the first place (as ensured by GitMacheteVisibilityPredicate)
+    List<GitRepository> repositories = List.ofAll(GitUtil.getRepositories(project));
```

Comment:
I'd put a comment here similar to what's in `com.virtuslab.gitmachete.frontend.ui.impl.table.GitMacheteGraphTableManager#queueGraphTableRefreshOnDispatchThread`: you're accessing git repo list on UI thread, but that's likely still okay